src/components/DropZone/components/FileUpload/FileUpload.scss

```diff
@@ -1,12 +1,61 @@
 @import '../../../../styles/common';
 
 $fileupload-padding: rem(15px);
+$slim-min-height: rem(30px);
+$slim-vertical-padding: ($slim-min-height - line-height(body) - rem(2px)) / 2;
 
 .FileUpload {
   padding: $fileupload-padding;
   text-align: center;
 }
 
+.Button {
+  @include button-base;
+
+  &.disabled {
+    @include base-button-disabled;
+    cursor: not-allowed;
+    box-shadow: none;
+  }
+
+  &.focused {
+    border-color: color('indigo');
+    outline: 0;
+    box-shadow: 0 0 0 1px color('indigo');
+
+    @include high-contrast-button-outline;
+  }
+
+  .sizeSlim {
+    min-height: $slim-min-height;
+    padding: $slim-vertical-padding spacing(base-tight);
+  }
+}
+
+.ActionTitle {
+  color: color('blue');
+  text-decoration: none;
+
+  &:not(.ActionTitle-disabled) {
+    cursor: pointer;
+
+    &:hover,
+    &:active {
+      color: color('blue', 'dark');
+      text-decoration: underline;
+    }
+  }
+}
+
+.ActionTitle-focused {
+  color: color('blue', 'dark');
+  text-decoration: underline;
+}
+
+.ActionTitle-disabled {
+  color: color('ink', 'lightest');
+}
```

Comment:
I typically avoid `:not()` unless absolutely needed. `:not()` communicates intent correctly but also adds an additional layer of state management.

(this is an opinion, not something I would say I'm definitely right about)

Suggestion: remove `&:not(.ActionTitle-disabled) {`, add `pointer-events: none;` to `.ActionTitle-disabled`.

```scss
.ActionTitle {
  color: color('blue');
  text-decoration: none;
  cursor: pointer;

  &:hover,
  &:active {
    color: color('blue', 'dark');
    text-decoration: underline;
  }
}

.ActionTitle-focused {
  color: color('blue', 'dark');
  text-decoration: underline;
}

.ActionTitle-disabled {
  color: color('ink', 'lightest');
  pointer-events: none;
}
```

Reply:
In this particular case I went with the `:not()`  to avoid having to override the hover and active on both `-focused` and `-disabled` and from having to deal with the double classes on `.ActionTitle-focused.ActionTitle-disabled` or relying on the order of css. I find here the intent is clear, we have just the right amount of specificity to not have to rely on order and this will produce the least about of code if I'm not mistaken. Are you good with that?